ql/src/java/org/apache/hadoop/hive/ql/ddl/table/creation/ShowCreateTableOperation.java

```diff
@@ -73,198 +82,207 @@ public ShowCreateTableOperation(DDLOperationContext context, ShowCreateTableDesc
   public int execute() throws HiveException {
     // get the create table statement for the table and populate the output
     try (DataOutputStream outStream = DDLUtils.getOutputStream(new Path(desc.getResFile()), context)) {
-      return showCreateTable(outStream);
+      Table table = context.getDb().getTable(desc.getTableName(), false);
+      String command = table.isView() ?
+          getCreateViewCommand(table) :
+          getCreateTableCommand(table);
+      outStream.write(command.getBytes(StandardCharsets.UTF_8));
+      return 0;
+    } catch (IOException e) {
+      LOG.info("show create table: ", e);
+      return 1;
     } catch (Exception e) {
       throw new HiveException(e);
     }
   }
 
-  private int showCreateTable(DataOutputStream outStream) throws HiveException {
-    boolean needsLocation = true;
-    StringBuilder createTabCommand = new StringBuilder();
+  private static final String CREATE_VIEW_COMMAND = "CREATE VIEW `%s` AS %s";
 
-    Table tbl = context.getDb().getTable(desc.getTableName(), false);
-    List<String> duplicateProps = new ArrayList<String>();
-    try {
-      needsLocation = CreateTableOperation.doesTableNeedLocation(tbl);
+  private String getCreateViewCommand(Table table) {
+    return String.format(CREATE_VIEW_COMMAND, desc.getTableName(), table.getViewExpandedText());
+  }
 
-      if (tbl.isView()) {
-        String createTabStmt = "CREATE VIEW `" + desc.getTableName() + "` AS " + tbl.getViewExpandedText();
-        outStream.write(createTabStmt.getBytes(StandardCharsets.UTF_8));
-        return 0;
-      }
+  private static final String CREATE_TABLE_TEMPLATE =
+      "CREATE <" + TEMPORARY + "><" + EXTERNAL + ">TABLE `<" + NAME + ">`(\n" +
+      "<" + LIST_COLUMNS + ">)\n" +
+      "<" + COMMENT + ">\n" +
+      "<" + PARTITIONS + ">\n" +
+      "<" + BUCKETS + ">\n" +
+      "<" + SKEWED + ">\n" +
+      "<" + ROW_FORMAT + ">\n" +
+      "<" + LOCATION_BLOCK + ">" +
+      "TBLPROPERTIES (\n" +
+      "<" + PROPERTIES + ">)\n";
 
-      createTabCommand.append("CREATE <" + TEMPORARY + "><" + EXTERNAL + ">TABLE `");
-      createTabCommand.append(desc.getTableName() + "`(\n");
-      createTabCommand.append("<" + LIST_COLUMNS + ">)\n");
-      createTabCommand.append("<" + TBL_COMMENT + ">\n");
```

Comment:
It seems we have removed some of the information we were printing, e.g., `TBL_COMMENT` or `TBL_LOCATION`. Is this because they will be included in the table properties?

Reply:
They are still there, just removed the unnecessary TBL_ prefix - they are all for the table :)